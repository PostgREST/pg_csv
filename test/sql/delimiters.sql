-- semicolon delimiter
SELECT csv_agg(x, ';') AS body
FROM   projects x;

-- pipe delimiter
SELECT csv_agg(x, '|') AS body
FROM   projects x;

-- tab delimiter
SELECT csv_agg(x, E'\t') AS body
FROM   projects x;

-- newline is forbidden as delimiter
SELECT csv_agg(x, E'\n') AS body
FROM   projects x;

-- double quote is forbidden as delimiter
SELECT csv_agg(x, '"') AS body
FROM   projects x;

-- carriage return is forbidden as delimiter
SELECT csv_agg(x, E'\r') AS body
FROM   projects x;
