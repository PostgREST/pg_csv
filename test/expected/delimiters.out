-- semicolon delimiter
SELECT csv_agg(x, ';') AS body
FROM   projects x;
             body              
-------------------------------
 id;name;client_id            +
 1;Windows 7;1                +
 2;has,comma;1                +
 ;;                           +
 4;OSX;2                      +
 ;"has""quote";               +
 5;"has,comma and ""quote""";7+
 6;"has                       +
  LF";7                       +
 7;"has \r CR";8              +
 8;"has \r                    +
  CRLF""";8
(1 row)

-- pipe delimiter
SELECT csv_agg(x, '|') AS body
FROM   projects x;
             body              
-------------------------------
 id|name|client_id            +
 1|Windows 7|1                +
 2|has,comma|1                +
 ||                           +
 4|OSX|2                      +
 |"has""quote"|               +
 5|"has,comma and ""quote"""|7+
 6|"has                       +
  LF"|7                       +
 7|"has \r CR"|8              +
 8|"has \r                    +
  CRLF"""|8
(1 row)

-- tab delimiter
SELECT csv_agg(x, E'\t') AS body
FROM   projects x;
                   body                    
-------------------------------------------
 id      name    client_id                +
 1       Windows 7       1                +
 2       has,comma       1                +
                                          +
 4       OSX     2                        +
         "has""quote"                     +
 5       "has,comma and ""quote"""       7+
 6       "has                             +
  LF"    7                                +
 7       "has \r CR"     8                +
 8       "has \r                          +
  CRLF"""        8
(1 row)

-- newline is forbidden as delimiter
SELECT csv_agg(x, E'\n') AS body
FROM   projects x;
ERROR:  delimiter cannot be newline, carriage return or double quote
-- double quote is forbidden as delimiter
SELECT csv_agg(x, '"') AS body
FROM   projects x;
ERROR:  delimiter cannot be newline, carriage return or double quote
-- carriage return is forbidden as delimiter
SELECT csv_agg(x, E'\r') AS body
FROM   projects x;
ERROR:  delimiter cannot be newline, carriage return or double quote
